src/components/SubscriptionCard.js

```diff
@@ -60,22 +63,37 @@ class SubscriptionCard extends React.Component {
     });
   }
 
-  async update(paymentMethod) {
+  async update({ paymentMethod, amount }) {
     const { intl } = this.props;
     this.setState({ loading: true });
-    let res;
     try {
-      res = await this.props.updateSubscription(this.props.subscription.id, paymentMethod);
+      await this.props.updateSubscription(this.props.subscription.id, paymentMethod, amount);
+      if (amount) {
+        // the card is gonna be unmounted, calling setState would result in a React warning
+        return;
```

Comment:
Have you tested error states? For example, when I enter "0" and click "Update", it says "Updated" and nothing changes. Would be best to gray out `Update` button until value is greater than 1. Or pass the correct to the user.

Reply:
I'll check.